src/ui/identities/update_identity_payout_address.rs

```diff
@@ -0,0 +1,239 @@
+            ui.add_space(5.0);
+            ui.push_id("funding_combo_id", |ui| {
+                ComboBox::from_label("")
+                    .selected_text(
+                        self.selected_funding_address
+                            .as_ref() // Get a reference to the Option<Address>
+                            .map(|address| address.to_string()) // Convert Address to String
+                            .unwrap_or_else(|| "".to_string()), // Use default "" if None
+                    )
+                    .show_ui(ui, |ui| {
+                        for (address, _) in &wallet.known_addresses {
+                            if ui
+                                .selectable_value(
+                                    &mut self.selected_funding_address,
+                                    Some(address.clone()),
+                                    address.to_string(),
+                                )
+                                .clicked()
+                            {}
+                        }
+                    });
+            });
+            ui.add_space(20.0);
+            if let Some(selected_address) = &self.selected_funding_address {
+                if let Some(value) = wallet.address_balances.get(&selected_address) {
+                    ui.label(format!("Selected Address has a balance of {} DASH", value));
+                } else {
+                    // TODO: Why sometimes balance is not found?
+                    //ui.label("Balance NOT FOUND DASH".to_string());
+                }
+            }
+        }
+    }
+}
+
+impl ScreenLike for UpdateIdentityPayoutScreen {
+    fn display_message(&mut self, message: &str, _message_type: MessageType) {
+        if _message_type == MessageType::Error {
+            self.error_message = Some(message.to_string());
+        }
+    }
+
+    /// Renders the UI components for the withdrawal screen
+    fn ui(&mut self, ctx: &Context) -> AppAction {
+        let mut action = add_top_panel(
+            ctx,
+            &self.app_context,
+            vec![
+                ("Identities", AppAction::GoToMainScreen),
+                ("Update Payout Address", AppAction::None),
+            ],
+            vec![],
+        );
+
+        egui::CentralPanel::default().show(ctx, |mut ui| {
+            if (self.identity.identity_type == IdentityType::User) {
+                ui.heading(
+                    "Updating Payout Address for User identities is not allowed.".to_string(),
+                );
+                return;
+            }
+            if (!self.app_context.has_wallet.load(Ordering::Relaxed)) {
+                ui.heading("Load a Wallet in order to continue.".to_string());
+                return;
+            }
+            ui.heading("Update Payout Address".to_string());
+            ui.add_space(20.0);
+
+            ui.heading("Load Address from wallet".to_string());
+            self.render_wallet_selection(&mut ui);
```

Comment:
I think you should render wallet unlock here as well after a wallet is selected, otherwise there will be an error when they try to update